test/TestApps/PushNotificationsTestApp/main.cpp

```diff
@@ -24,49 +25,328 @@ void SignalPhase(const std::wstring& phaseEventName)
+        hr = channelRequestException.code();
+    }
+
+    if (hr == E_INVALIDARG)
+        return true;
+    else
+        return false;
 }
 
-void sendRequestToServer(winrt::hstring channel)
+bool ChannelRequestUsingRemoteId()
 {
-    HttpResponseMessage httpResponseMessage;
-    std::wstring httpResponseBody;
-    try {
-        // Construct the HttpClient and Uri. This endpoint is for test purposes only.
-        HttpClient httpClient;
-        Uri requestUri{ L"http://localhost:7071/api/PostPushNotification" };
-
-        // Construct the JSON to post.
-        HttpStringContent jsonContent(
-            BuildNotificationPayload(channel.c_str()),
-            UnicodeEncoding::Utf8,
-            L"application/json");
-
-        // Post the JSON, and wait for a response.
-        httpResponseMessage = httpClient.PostAsync(
-            requestUri,
-            jsonContent).get();
-
-        // Make sure the post succeeded, and write out the response.
-        httpResponseMessage.EnsureSuccessStatusCode();
-        httpResponseBody = httpResponseMessage.Content().ReadAsStringAsync().get();
-        std::wcout << httpResponseBody.c_str() << std::endl;
+    // winrt::guid remoteId { L"a2e4a323-b518-4799-9e80-0b37aeb0d225" };
+    wil::unique_handle channelEvent = wil::unique_handle(CreateEvent(nullptr, FALSE, FALSE, nullptr));
+    auto hr = S_OK;
+    auto channelOperation = PushNotificationManager::CreateChannelAsync(winrt::guid(0xa2e4a323, 0xb518, 0x4799, { 0x9e, 0x80, 0x0b, 0x37, 0xae, 0xb0, 0xd2, 0x25 }));
+
+    channelOperation.Progress(
+        [](
+            IAsyncOperationWithProgress<PushNotificationCreateChannelResult, PushNotificationCreateChannelStatus> const& /* sender */,
+            PushNotificationCreateChannelStatus const& args)
+        {
+            if (args.status == PushNotificationChannelStatus::InProgress)
+            {
+                // This is basically a noop since it isn't really an error state
+                std::cout << "The channel request is in progress " << args.extendedError << std::endl;
+            }
+            else if (args.status == PushNotificationChannelStatus::InProgressRetry)
+            {
+                std::cout << "The channel request is in back-off retry mode because of a retryable error code: " << args.extendedError << " Expect delays in acquiring it" << std::endl;
+            }
+        });
+
+    // Setup the completed event handler
+    channelOperation.Completed(
+        [&channelEvent](
+            IAsyncOperationWithProgress<PushNotificationCreateChannelResult, PushNotificationCreateChannelStatus> const& sender,
+            AsyncStatus const /* asyncStatus */)
+        {
+            auto result = sender.GetResults();
+            if (result.Status() == PushNotificationChannelStatus::CompletedSuccess)
+            {
+                auto channelUri = result.Channel().Uri();
+                auto channelExpiry = result.Channel().ExpirationTime();
```

Comment:
Should we print this or do something more than just calling the API? It is not used.

Reply:
Actually std::cout are unneccessary, got rid of them. Unit test has to tell us whether it is a success/failure.